client/src/components/map.tsx

```diff
@@ -7,101 +7,141 @@ import VectorLayer from 'ol/layer/Vector';
 import VectorSource from 'ol/source/Vector';
 import {fromLonLat} from 'ol/proj';
 import * as styles from './map.module.scss';
-import olms from 'ol-mapbox-style';
+import TileLayer from 'ol/layer/Tile';
+import VectorTileLayer from 'ol/layer/VectorTile.js';
+import VectorTileSource from 'ol/source/VectorTile.js';
+import MVT from 'ol/format/MVT.js';
+import {Fill, Style} from 'ol/style.js';
+import XYZ from 'ol/source/XYZ';
+import * as d3 from 'd3';
 
-interface IMapWrapperProps {
-  features: Feature<Geometry>[],
-};
+import Overlay from 'ol/Overlay';
 
-const mapConfig = {
-  'version': 8,
-  'cursor': 'pointer',
-  'sources': {
-    'carto-light': {
-      'type': 'raster',
-      'tiles': [
-        'https://a.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png',
-        'https://b.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png',
-        'https://c.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png',
-        'https://d.basemaps.cartocdn.com/light_all/{z}/{x}/{y}@2x.png',
-      ],
-    },
-    'custom': {
-      'projection': 'EPSG:3857',
-      'type': 'vector',
-      'tiles': [
-        'https://gis.data.census.gov/arcgis/rest/services/Hosted/VT_2019_150_00_PY_D1/VectorTileServer/tile/{z}/{y}/{x}.mvt',
-      ],
-    },
-  },
-  'layers': [
-    {
-      'id': 'carto-light-layer',
-      'source': 'carto-light',
-      'type': 'raster',
-      'minzoom': 0,
-      'maxzoom': 22,
-    },
-    {
-      'id': 'blocks',
-      'type': 'line',
-      'source': 'custom',
-      'source-layer': 'BlockGroup',
-      'minzoom': 0,
-      'layout': {
-        'line-cap': 'round',
-        'line-join': 'round',
-      },
 
-      'paint': {
-        'line-opacity': 0.6,
-        'line-color': 'red',
-        'line-width': 1,
-      },
-    },
-  ],
+interface IMapWrapperProps {
+  features: Feature<Geometry>[],
 };
 
-
 // The below adapted from
 // https://taylor.callsen.me/using-openlayers-with-react-functional-components/
 const MapWrapper = ({features}: IMapWrapperProps) => {
   const [map, setMap] = useState<Map>();
+  const [currentOverlay, setCurrentOverlay] = useState<Overlay>();
   const [featuresLayer, setFeaturesLayer] = useState<VectorLayer>();
+  const [selectedFeature, setSelectedFeature] = useState<Feature>();
+  const [hoveredFeature, setHoveredFeature] = useState<Feature>();
 
   const mapElement = useRef() as
-        React.MutableRefObject<HTMLInputElement>;
+    React.MutableRefObject<HTMLInputElement>;
+  const mapRef = useRef() as React.MutableRefObject<Map>;
+
 
-  useEffect( () => {
+  const popupContainer = React.useRef<HTMLDivElement>(null);
+  const popupCloser = React.useRef<HTMLDivElement>(null);
+  const popupContent = React.useRef<HTMLDivElement>(null);
+
+  const overlayRef = useRef() as React.MutableRefObject<Map>;
+  mapRef.current = map;
```

Comment:
I'm getting some typescript complaints from these two lines.
> TS2322: Type 'Map | undefined' is not assignable to type 'Map'.   Type 'undefined' is not assignable to type 'Map'.

I think you just need to do a check that `map` is not null. e.g `if (map) {`

Same for `currentOverlay`